zenpy/lib/api.py

```diff
@@ -1821,38 +1821,110 @@ def create(self, section):
                                                     attachment=attachment,
                                                     inline=inline,
                                                     file_name=file_name,
                                                     content_type=content_type)
 
-    def create_unassociated(self, attachment, inline=False):
+    def create_unassociated(self, attachment, inline=False, file_name=None, content_type=None):
+        """
+        You can use this endpoint for bulk imports.
+        It lets you upload a file without associating it to an article until later.
+        Check Zendesk documentation `important notes
+        <https://developer.zendesk.com/rest_api/docs/help_center/article_attachments#create-unassociated-attachment>
+
+        :param attachment: File object or os path to file
+        :param inline: If true, the attached file is shown in the dedicated admin UI
+            for inline attachments and its url can be referenced in the HTML body of
+            the article. If false, the attachment is listed in the list of attachments.
+            Default is `false`
+        :param file_name: you can set filename on file upload.
+        :param content_type: The content type of the file. `Example: image/png`, Zendesk can ignore it.
+        :return: :class:`ArticleAttachment` object
+        """
         return HelpdeskAttachmentRequest(self).post(self.endpoint.create_unassociated,
                                                     attachment=attachment,
-                                                    inline=inline)
+                                                    inline=inline,
+                                                    file_name=file_name,
+                                                    content_type=content_type)
 
     @extract_id(ArticleAttachment)
     def delete(self, article_attachment):
+        """
+        This function completely wipes attachment from Zendesk Helpdesk article.
+
+        :param article_attachment: :class:`ArticleAttachment` object or numeric article attachment id.
+        :return: status_code == 204 on success
+        """
         return HelpdeskAttachmentRequest(self).delete(self.endpoint.delete, article_attachment)
 
+    @extract_id(Article)
+    def bulk_attachments(self, article, attachments):
+        """
+        This function implements associating attachments to an article after article creation (for
+        unassociated attachments).
+
+        :param article: Article id or :class:`Article` object
+        :param attachments: class:`ArticleAttachment` object, or list of :class:`ArticleAttachment` objects,
+        up to 20 supported. `Zendesk documentation.
+        <https://developer.zendesk.com/rest_api/docs/help_center/articles#associate-attachments-in-bulk-to-article>`__
+        :return:
+        """
+        return HelpdeskAttachmentRequest(self).bulk_associate(self.endpoint.bulk_attachments, article=article,
```

Comment:
The pattern with these request classes is to use the methods post/put/delete. Instead of calling the `bulk_associate` method directly here, it would be more consistent to pass a flag to the `post` method and handle it within. Not a big deal, but keeps the pattern the same as the other request objects.